src/main/java/pink/zak/minestom/towerdefence/ui/spawner/TroopSpawnerUI.java

```diff
@@ -71,8 +71,8 @@ public TroopSpawnerUI(@NotNull MobStorage mobStorage, @NotNull GameUser gameUser
             // this is not the player who opened the inventory, fast exit
             if (!p.equals(gameUser.getPlayer())) return;
 
-            // check if slots are within the tab range
-            if (this.tab != null && slot >= 9 && slot < 27) {
+            // check if slots are within the tab range and click item is NOT air
+            if (this.tab != null && slot >= 9 && slot < 27 && !result.getClickedItem().isAir()) {
                 this.tab.onClick(slot - 9);
                 return;
             }
```

Comment:
Wondering the intention for this change? Doing this will direct clicks to the outer UI rather than the inner when the clicked item is air.